### PR TITLE
Salt generation

### PR DESCRIPTION
**Pull request by ThatDamnPineapple**
https://trello.com/c/37mSaaNU/129-salt-1, makes seasalt and table salt both placeable, as well as making seasalt generate.

Only thing left to do is MAYBE to make it generate underground?

### CODE REVIEW

**Comment by ThatDamnPineapple** [comment]
Made it generate underground

File: Content/Tiles/Cooking/Salts.cs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+		public override void AI()
+		{
+			if (Main.rand.NextBool(5))
+			{
+				int dust = Dust.NewDust(Projectile.position, Projectile.width, Projectile.height, DustType);
+				Main.dust[dust].velocity.X *= 0.4f;
+			}
+
+			Projectile.tileCollide = true;
+			Projectile.localAI[1] = 0f;
```
Comment (ScalarVector1) [inline Content/Tiles/Cooking/Salts.cs:182]: We should be creating ref properties to give these fields readable names, Additionally we should use custom variables as opposed to localAI as localAI provides no benefits

File: Content/WorldGeneration/GenerateSeaSalt.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			progress.Message = "Adding Salt";
+			PlaceSeaSalt(20, 400);
+			PlaceSeaSalt(Main.maxTilesX - 400, Main.maxTilesX - 20);
+		}
+
+		private void PlaceSeaSalt(int xStart, int xEnd)
+		{
+			for (int i = xStart; i < xEnd; i++)
+			{
+				int j = 100;
```
Comment (ScalarVector1) [inline Content/WorldGeneration/GenerateSeaSalt.cs:21]: Is there any reason this is outside of the for loop?

File: Content/WorldGeneration/GenerateSeaSalt.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+				for (; j < Main.maxTilesY; j++)
+				{
+					Tile tile = Framing.GetTileSafely(i, j);
+
+					if (tile.HasTile && Main.tileSolid[tile.TileType])
+					{
+						if (!Main.rand.NextBool(60) || j > Main.rockLayer)
+							continue;
```
Comment (ScalarVector1) [inline Content/WorldGeneration/GenerateSeaSalt.cs:29]: linebreak here would likely aid readability

File: Content/WorldGeneration/PermafrostGen.cs [hunk 4]
```diff
@@ -120,9 +120,9 @@ public void PermafrostGen(GenerationProgress progress, GameConfiguration configu
 			{
 				if (Main.tile[center, y].TileType == TileID.SnowBlock || Main.tile[center, y].TileType == TileID.IceBlock)
 				{
-					StructureHelper.Generator.GenerateStructure("Structures/TouchstoneAltar", new Point16(center, y - 12), Mod);
-					(TileEntity.ByPosition[new Point16(center + 9, y - 12 + 6)] as TouchstoneTileEntity).targetPoint = oldPos;
-					break;
+					//StructureHelper.Generator.GenerateStructure("Structures/TouchstoneAltar", new Point16(center, y - 12), Mod);
```
Comment (ScalarVector1) [inline Content/WorldGeneration/PermafrostGen.cs:123]: Please revert this change, this likely should not have been comitted

**Review by ScalarVector1 — CHANGES_REQUESTED** [review]
